broker/src/tunneldigger_broker/broker.py

```diff
@@ -77,6 +83,26 @@ def create_tunnel(self, broker, address, uuid, remote_tunnel_id, client_features
         else:
             tunnel_str = "(%s)" % uuid
 
+        # Rate limit creation of new tunnels with the same IP address.
+        # Runs broker.tunnel-connection-threshold hook if threshold exceeded:
+        if self.connection_rate_limit_per_ip_count > 0 and self.connection_rate_limit_per_ip_time > 0:
+            if address[0] not in self.last_tunnel_created_per_ip:
+                self.last_tunnel_created_per_ip[address[0]] = deque([now]) # Create deque collection if IP address does not exist in dict
+            if len(self.last_tunnel_created_per_ip[address[0]])-1 is self.connection_rate_limit_per_ip_count:
```

Comment:
Could you bind `self.last_tunnel_created_per_ip[address[0]]` to a variable? It gets used quite a bit here.

Reply:
I am also a bit confused by the `-1` here. If these are indeed all *previous* attempt, I think it should be just an equality comparison? At least, if you didn't pre-add the `now`.